utils/DeployedContractsHelpers.ts

```diff
@@ -151,91 +151,40 @@ export async function getBDStableWethPool(hre: HardhatRuntimeEnvironment, symbol
+  const tokenAAddress = await getContratAddress(hre, tokenASymbol);
+  const tokenBAddress = await getContratAddress(hre, tokenBSymbol);
+  const poolKey = getPoolKey(tokenAAddress, tokenBAddress, tokenASymbol, tokenBSymbol);
 
   const deployer = await getDeployer(hre);
   const stakingRewardsContract = await hre.ethers.getContractOrNull(`StakingRewards_${poolKey}`, deployer);
   return stakingRewardsContract && (stakingRewardsContract as StakingRewards);
 }
 
 export async function getAllBDStableStakingRewards(hre: HardhatRuntimeEnvironment): Promise<StakingRewards[]> {
-  const deployer = await getDeployer(hre);
-  const bdstablesSymbols = getAllBDStablesSymbols();
-  const bdx = await getBdx(hre);
   const stakingRewards: StakingRewards[] = [];
-  const stakingRewardsBdStablesMap = new Set<string>();
-
-  const bdxWethStakingRewards = await getStakingRewardsWithWeth(hre, await bdx.symbol());
-  if (!bdxWethStakingRewards) {
-    throw new Error("BDX-WETH staking reward is missing");
-  }
-  stakingRewards.push(bdxWethStakingRewards);
-
-  const bdxWbtcStakingRewards = await getStakingRewardsWithWbtc(hre, await bdx.symbol());
-  if (!bdxWbtcStakingRewards) {
-    throw new Error("BDX-WBTC staking reward is missing");
-  }
-  stakingRewards.push(bdxWbtcStakingRewards);
-
-  for (const symbolA of bdstablesSymbols) {
-    const wethStakingRewards = await getStakingRewardsWithWeth(hre, symbolA);
-    if (wethStakingRewards) {
-      stakingRewards.push(wethStakingRewards);
-    }
-
-    const wbtcStakingRewards = await getStakingRewardsWithWbtc(hre, symbolA);
-    if (wbtcStakingRewards) {
-      stakingRewards.push(wbtcStakingRewards);
-    }
-
-    const bdxStakingRewards = await getStakingRewardsWithBdx(hre, symbolA);
-    if (bdxStakingRewards) {
-      stakingRewards.push(bdxStakingRewards);
-    }
+  const supportedStakingPools = getListOfSupportedLiquidityPools(hre.network.name).filter(lp => lp.hasStakingPool);
 
-    for (const symbolB of bdstablesSymbols) {
-      const stableAddress = await getContratAddress(hre, symbolA);
-      const stableBAddress = await getContratAddress(hre, symbolB);
-      const poolKey = getPoolKey(stableBAddress, stableAddress, symbolB, symbolA);
-
-      // Do not repeat the same staking rewards twice
-      if (symbolA !== symbolB && !stakingRewardsBdStablesMap.has(poolKey)) {
-        const stablesStakingRewards = await hre.ethers.getContractOrNull(`StakingRewards_${poolKey}`, deployer);
-        if (stablesStakingRewards) {
-          stakingRewards.push(stablesStakingRewards as StakingRewards);
-          stakingRewardsBdStablesMap.add(poolKey);
-        }
+  await Promise.all(
+    supportedStakingPools.map(async stakingPool => {
+      const stakingReward = await getBDStableStakingReward(hre, stakingPool.tokenA, stakingPool.tokenB);
+      if (!stakingReward) {
+        throw new Error(`${stakingPool.tokenA}-${stakingPool.tokenB} staking reward is missing`);
```

Comment:
```suggestion
        throw new Error(`${stakingPool.tokenA}_${stakingPool.tokenB} staking reward is missing`);
```

That's the format we're using I believe